### PR TITLE
ROC-6119 Add resume payment for citizen endpoint

### PR DESCRIPTION
**Pull request by simad**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-6119

### Change description ###

Adding endpoint in claimstore to resume a payment 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataServiceTest.java [hunk 1]
```diff
@@ -252,17 +259,56 @@ public void initiatePaymentShouldReturnPaymentNextUrl() {
...
+                .withPayment(
+                    SamplePayment.builder()
+                        .status(PaymentStatus.SUCCESS)
+                        .nextUrl("http://payment.nexturl.test").build()
+                ).build(),
+            "000MC001"
+        );
+
+        when(caseMapper.from(any(CCDCase.class))).thenReturn(claim);
+        when(caseDetailsConverter.extractClaim(any((CaseDetails.class)))).thenReturn(claim);
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataServiceTest.java]: There are extra parentheses here

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java [hunk 2]
```diff
@@ -65,19 +69,20 @@
...
     private static final String SUBMITTING_CMC_INITIATE_PAYMENT_DESCRIPTION = "Submitting CMC initiate payment";
+    private static final String SUBMITTING_CMC_RESUME_PAYMENT_DESCRIPTION = "Submitting CMC resume payment";
 
     private static final String CCD_UPDATE_FAILURE_MESSAGE
         = "Failed updating claim in CCD store for case id %s on event %s";
 
     private static final String CCD_STORING_FAILURE_MESSAGE
         = "Failed storing claim in CCD store for case id %s on event %s";
 
+    @Value("${payments.returnUrl}") String returnUrl;
```
[the remark this answers is not in the document]
Comment (simad) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java]: 👍 I think I originally had it as pure frontendUrl and then building the return url on the fly - then I realised I could just put the returnUrl itself in the config, but forgot to make it final :D cheers

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataServiceTest.java (hunk 1 above)
Comment (simad) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataServiceTest.java]: I am a generous person!

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/PaymentStatus.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.Arrays;
+
+@Getter
+@AllArgsConstructor
+public enum PaymentStatus {
+    INITIATED("Initiated"),
+    SUCCESS("Success"),
+    FAILED("Failed"),
+    PENDING("Pending"),
+    DECLINED("Declined");
```
Comment (ckalista) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/PaymentStatus.java]: Heya - are we confident we have got an exhaustive list of statuses here?

Comment (simad) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/PaymentStatus.java]: yep, I've taken that from the ccpay codebase itself :)

**Pull request merged**